compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2213,16 +2213,20 @@ class Typer extends Namer
             traverse(xtree :: rest, facts)
           case none =>
             import untpd.modsDeco
-            val ctx1 = if (ctx.explicitNulls) {
-              mdef match {
-                case mdef: untpd.ValDef if ctx.owner.is(Method) && !mdef.mods.isOneOf(Lazy | Implicit) =>
-                  ctx.fresh.addFlowFacts(facts)
-                case _ => ctx
-              }
-            } else {
-              ctx
+            val ctx2 = mdef match {
+              case mdef: untpd.ValDef
+                if ctx.explicitNulls && ctx.owner.is(Method) && !mdef.mods.isOneOf(Lazy | Implicit) =>
+                val ctx1 = ctx.fresh.addFlowFacts(facts)
+                val sym = ctx1.effectiveScope.lookup(mdef.name)
```

Comment:
Do we know why we need to look up the symbol like this, as opposed to getting it from the mdef?

Reply:
comment added